internal/ad/ad.go

```diff
@@ -340,7 +341,7 @@ func (ad *AD) ListUsers(ctx context.Context, active bool) (users []string, err e
 
 	cacheDir := ad.policiesCacheDir
 	if active {
-		cacheDir = ad.krb5CacheDir
+		cacheDir = filepath.Join(ad.krb5CacheDir, "tracking")
```

Comment:
So, here, we are not checking the symlinks are still valid, there is thus no advantaged compared to only readDir the cacheDir (and ignoring subdirectories).

So, I suggest we either:
* rely on the symlink, but in that case, it would be good to see if the destination still exists (and ignoring those that don't, with a test for it)
* keep the current behaviour, but then basing on `krb5CacheDir`